### PR TITLE
Soft deprecate AcquireTokenByUsernamePassword to use string

### PR DESCRIPTION
**Pull request by neha-bhargava**
Fixes # 
#2437 

**Changes proposed in this request**
Soft deprecate the AcquireTokenByUsernamePassword with SecureString parameter to take string as parameter

**Testing**

**Performance impact**

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/IPublicClientApplication.cs [hunk 1]
```diff
@@ -131,9 +132,31 @@ public partial interface IPublicClientApplication : IClientApplicationBase
         /// in order to override the default authority set at the application construction. Note that the overriding authority needs to be part
         /// of the known authorities added to the application construction.
         /// </remarks>
+        [Obsolete("This method is deprecated. Please use AcquireTokenByUsernamePassword(IEnumerable<string> scopes, string username, string password) instead.", true)]
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/IPublicClientApplication.cs:135]: Let's soft-deprecate, i.e. recommend people to use string, but not block them if they prefer to continue to use SecureString.

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/IPublicClientApplication.cs:135]: Reason: ".NET no longer recommends using SecureString and MSAL would anyway need to put the plaintext value of the password on the wire, as required by the OAuth protocol. See https://docs.microsoft.com/en-us/dotnet/api/system.security.securestring?view=net-5.0#remarks"

File: src/client/Microsoft.Identity.Client/PublicClientApplication.cs [hunk 2]
```diff
@@ -230,7 +230,7 @@ public bool IsBrokerAvailable()
         /// in order to override the default authority set at the application construction. Note that the overriding authority needs to be part
         /// of the known authorities added to the application construction.
         /// </remarks>
-        [Obsolete("This method is deprecated. Please use AcquireTokenByUsernamePassword(IEnumerable<string> scopes, string username, string password) instead.", true)]
+        [Obsolete("This method is deprecated. Please use AcquireTokenByUsernamePassword(IEnumerable<string> scopes, string username, string password) instead.", false)]
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/PublicClientApplication.cs:233]: nit: I remember @pmaytak read some guidelines about warnings and messages and they recommend against using "please".

**Review by bgavrilMS — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/PublicClientApplication.cs (hunk 2 above)
Comment (pmaytak) [reply src/client/Microsoft.Identity.Client/PublicClientApplication.cs:233]: Yea, I remember that too, although I can't find the related docs.

File: src/client/Microsoft.Identity.Client/IPublicClientApplication.cs [hunk 3]
```diff
@@ -131,9 +132,31 @@ public partial interface IPublicClientApplication : IClientApplicationBase
         /// in order to override the default authority set at the application construction. Note that the overriding authority needs to be part
         /// of the known authorities added to the application construction.
         /// </remarks>
+        [Obsolete("This method is deprecated. Please use AcquireTokenByUsernamePassword(IEnumerable<string> scopes, string username, string password) instead.", false)]
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/IPublicClientApplication.cs:135]:
```suggestion
        /// .NET no longer recommends using SecureString and MSAL puts the plaintext value of the password on the wire, as required by the OAuth protocol. See <see href="https://docs.microsoft.com/en-us/dotnet/api/system.security.securestring?view=net-6.0#remarks">SecureString documentation</see>.
        /// </remarks>
        [Obsolete("Using SecureString is not recommended. Use AcquireTokenByUsernamePassword(IEnumerable<string> scopes, string username, string password) instead.", false)]
```

File: src/client/Microsoft.Identity.Client/PublicClientApplication.cs [hunk 4]
```diff
@@ -228,10 +230,39 @@ public bool IsBrokerAvailable()
         /// in order to override the default authority set at the application construction. Note that the overriding authority needs to be part
         /// of the known authorities added to the application construction.
         /// </remarks>
+        [Obsolete("This method is deprecated. Please use AcquireTokenByUsernamePassword(IEnumerable<string> scopes, string username, string password) instead.", false)]
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/PublicClientApplication.cs:233]:
```suggestion
        /// .NET no longer recommends using SecureString and MSAL puts the plaintext value of the password on the wire, as required by the OAuth protocol. See <see href="https://docs.microsoft.com/en-us/dotnet/api/system.security.securestring?view=net-6.0#remarks">SecureString documentation</see>.
        /// </remarks>
        [Obsolete("Using SecureString is not recommended. Use AcquireTokenByUsernamePassword(IEnumerable<string> scopes, string username, string password) instead.", false)]
```

File: src/client/Microsoft.Identity.Client/PublicClientApplication.cs [hunk 5]
```diff
@@ -228,10 +230,39 @@ public bool IsBrokerAvailable()
...
+        /// See https://aka.ms/msal-net-up for details.
+        /// </summary>
+        /// <param name="scopes">Scopes requested to access a protected API</param>
+        /// <param name="username">Identifier of the user application requests token on behalf.
+        /// Generally in UserPrincipalName (UPN) format, e.g. <c>john.doe@contoso.com</c></param>
+        /// <param name="password">User password as a string.</param>
+        /// <returns>A builder enabling you to add optional parameters before executing the token request</returns>
+        /// <remarks>You can also pass optional parameters by chaining the builder with:
+        /// <see cref="AbstractAcquireTokenParameterBuilder{T}.WithExtraQueryParameters(Dictionary{string, string})"/> to pass
+        /// additional query parameters to the STS, and one of the overrides of <see cref="AbstractAcquireTokenParameterBuilder{T}.WithAuthority(string, bool)"/>
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/PublicClientApplication.cs:258]: nit:
```suggestion
        /// additional query parameters to the Azure AD, and one of the overrides of <see cref="AbstractAcquireTokenParameterBuilder{T}.WithAuthority(string, bool)"/>
```

**Review by pmaytak — APPROVED** [review]

**Pull request merged**